src/test/scala/com/quartethealth/spark/fixedwidth/FixedwidthSuite.scala

```diff
@@ -0,0 +1,117 @@
+package com.quartethealth.spark.fixedwidth
+
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.types.{DoubleType, IntegerType, StringType, StructField, StructType}
+import org.apache.spark.sql.{DataFrame, SQLContext}
+import org.scalatest.{BeforeAndAfterAll, FunSuite}
+
+
+class FixedwidthSuite extends FunSuite with BeforeAndAfterAll {
+  protected def fruit_resource(name: String = ""): String =
+    s"src/test/resources/fruit_${name}_fixedwidth.txt"
+
+  protected val fruitWidths = Array(3, 10, 5, 4)
+  protected val fruitSize = 7
+  protected val fruitFirstRow = Seq(56, "apple", "TRUE", 0.56)
+
+  protected val fruitSchema = StructType(Seq(
+    StructField("val", IntegerType),
+    StructField("name", StringType),
+    StructField("avail", StringType),
+    StructField("cost", DoubleType)
+  ))
+
+  protected var sqlContext: SQLContext = _
```

Comment:
you should be able to make this an immutable `lazy val` with the expression that you execute below, and it'll evaluate on first usage then not be modified otherwise.